base_business_document_import/models/business_document_import.py

```diff
@@ -107,6 +107,7 @@ def _match_partner(
             # use base_vat_sanitized
             partners = rpo.search(
                 domain + [
+                    ('company_id', '=', self.env.user.company_id.id),
```

Comment:
The problem is if this comes from sudo or in the ideal company-per-document approach, so don't we have a company pointer for the current document?

Reply:
We can test with `hasattr` if the model where is implemented have it, and if not, fallback to the one you propose.

Reply:
@pedrobaeza can you check if the latest commit is what you meant?

Reply:
Yes!